blaze/expr/tests/test_strings.py

```diff
@@ -28,3 +27,16 @@ def test_str_upper_schema(ds):
     assert (expr_upper.schema.measure ==
             expr_lower.schema.measure ==
             dshape('%sstring' % ('?' if '?' in ds else '')).measure)
+
```

Comment:
We should add tests for the following cases:
* Calling the functional form (rather than the method form) of `str_cat()` works and does proper error checking for both the lhs and rhs types / dshapes,
* test the resulting dshape for str_cat, specifically, what happens when strcat is passed in option or non-option string columns?